README.md

```diff
@@ -1 +1,1146 @@
+
+```
+   $ ballerina run message_transformation/
+```
+
+Observe the tracing using Jaeger UI using the following URL.
+
+```
+   http://localhost:16686
+```
+
+### Metrics
+Metrics and alerts are built-in with ballerina. We will use Prometheus as the monitoring tool. Follow the steps below to set up Prometheus and view metrics for 'passthrough'.
+
+You can add the following configurations for metrics. Note that these configurations are optional if you already have the basic configuration in `ballerina.conf` as described under the `Observability` section.
+
+```ballerina
+   [b7a.observability.metrics]
+   enabled=true
+   provider="micrometer"
+
+   [b7a.observability.metrics.micrometer]
+   registry.name="prometheus"
+
+   [b7a.observability.metrics.prometheus]
+   port=9700
+   hostname="0.0.0.0"
+   descriptions=false
+   step="PT1M"
+```
+
+Create a file `prometheus.yml` inside the `/tmp/` location. Add the below configurations to the `prometheus.yml` file.
+
+```
+   global:
+     scrape_interval:     15s
+     evaluation_interval: 15s
+
+   scrape_configs:
+     - job_name: prometheus
+       static_configs:
+         - targets: ['172.17.0.1:9797']
+```
+
+> **NOTE**: Replace `172.17.0.1` if your local Docker IP differs from `172.17.0.1`
+   
+Run the Prometheus docker image using the following command
+
+```
+   $ docker run -p 19090:9090 -v /tmp/prometheus.yml:/etc/prometheus/prometheus.yml \
+   prom/prometheus
+```
+   
+You can access Prometheus at the following URL.
+
+```
+   http://localhost:19090/
+```
+
+> **NOTE**: Ballerina will by default have the following metrics for HTTP server connector. You can enter the following expression in Prometheus UI.
+>    -  http_requests_total
+>    -  http_response_time
+
+### Logging
+Ballerina has a log package for logging to the console. You can import `ballerina/log` package and start logging. The following section describes how to search, analyze, and visualize logs in real time using Elastic Stack.
+
+Start the Ballerina service with the following command from `eip-message-transformation/guide`
+
+```
+   $ nohup ballerina run message_transformation/ &>> ballerina.log&
+```
+
+> **NOTE**: This writes the console log to the `ballerina.log` file in the `eip-message-transformation/guide` directory.
+
+Start Elasticsearch using the following command.
+
+```
+   $ docker run -p 9200:9200 -p 9300:9300 -it -h elasticsearch --name \
+   elasticsearch docker.elastic.co/elasticsearch/elasticsearch:6.2.2 
+```
+
+> **NOTE**: Linux users might need to run `sudo sysctl -w vm.max_map_count=262144` to increase `vm.max_map_count`.
+   
+Start Kibana plugin for data visualization with Elasticsearch.
+
+```
+   $ docker run -p 5601:5601 -h kibana --name kibana --link \
+   elasticsearch:elasticsearch docker.elastic.co/kibana/kibana:6.2.2     
+```
+
+Configure logstash to format the Ballerina logs.
+
+i) Create a file named `logstash.conf` with the following content.
+
+```
+input {  
```

Comment:
Can we add some comments or more explanation on what this code does? The user needs to understand exactly why they are doing each step.